### ISSUE TITLE
Analyze this content and decided whether to keep it

### ISSUE THREAD

**Issue by mairaw** [dotnet/docs#8121]
This section seems to go over some of the threading classes. Details about the classes should belong in the reference docs and we could perhaps eliminate some of these articles. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ca22a9e5-5a43-df98-43b9-c484e30b7baf
* Version Independent ID: 23243e79-4cd8-ea3a-1cdb-97a0ede877d4
* Content: [Threading objects and features](https://docs.microsoft.com/en-us/dotnet/standard/threading/threading-objects-and-features)
* Content Source: [docs/standard/threading/threading-objects-and-features.md](https://github.com/dotnet/docs/blob/master/docs/standard/threading/threading-objects-and-features.md)
* Product: **dotnet**
* GitHub Login: @rpetrusha
* Microsoft Alias: **ronpet**

**Comment by pkulikov** [comment]
@mairaw I had the same thoughts when worked on #7790 

Some topics might be eliminated and redirected to their analogs in the API reference as is. Other topics can be eliminated after their API "twins" are updated. The following articles are the candidates:

- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/mutexes
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/reader-writer-locks
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/semaphore-and-semaphoreslim
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/interlocked-operations
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/autoresetevent
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/manualresetevent-and-manualreseteventslim
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/eventwaithandle
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/countdownevent
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/barrier
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/spinlock
- [ ] https://docs.microsoft.com/en-us/dotnet/standard/threading/spinwait

I'd be happy to contribute to fix this issue. A PR per one or two articles looks like a good way to proceed. 

@rpetrusha what do you think?

Related to #3819

**Comment by rpetrusha** [comment]
I agree, @pkulikov.  Most of this content, I think, can be moved to the API reference, where it's far more discoverable than here.

**Comment by pkulikov** [comment]
>Most of this content, I think, can be moved to the API reference, where it's far more discoverable than here

Cool! @rpetrusha please assign this issue to me, then.

**Comment by mairaw** [comment]
Done. Thanks @pkulikov!

**Comment by pkulikov** [comment]
I need to admit I have no time to work on this issue and, thus, unassign.
